app-backend/api/src/main/scala/user/Routes.scala

```diff
@@ -82,8 +82,8 @@ trait UserRoutes extends Authentication
   }
 
   def updateOwnUser: Route = authenticate { user =>
-    entity(as[User]) { updatedUser =>
```

Comment:
I think this should still be the `User` object -- there are other aspects of themselves that users should be able to update that include:
 - planet credential
 - visibility
 - email notifications

The other aspects of a user should be sanitized and a user should not be able to update:
 - if they are active or not
 - if they are a superuser
 - role
 - modified/creation times

Reply:
I think this is a fine comment to make, but the original also restricted the update scope to planet token only. I'm not 100% sure why we even have the special credential update methods, but :man_shrugging: Maybe this should be rescoped to getting rid of those.

Reply:
Ok yeah -- we just talked about it --

This PR should kill uses of `storePlanetAccessToken` and `storeDropboxAccessToken` and just use the `UserDao.updateUser` method to perform the update (and make sure that succeeds).

Sorry for the rescope -- I don't think we knew about the terrible choice I made a while ago when we made this issue.